src/main/java/it/gov/pagopa/pu/debtpositions/mapper/PaymentOptionMapper.java

```diff
@@ -0,0 +1,56 @@
+package it.gov.pagopa.pu.debtpositions.mapper;
+
+import it.gov.pagopa.pu.debtpositions.dto.Installment;
+import it.gov.pagopa.pu.debtpositions.dto.InstallmentPIIDTO;
+import it.gov.pagopa.pu.debtpositions.dto.generated.PaymentOptionDTO;
+import it.gov.pagopa.pu.debtpositions.enums.PaymentOptionType;
+import it.gov.pagopa.pu.debtpositions.model.InstallmentNoPII;
+import it.gov.pagopa.pu.debtpositions.model.PaymentOption;
+import org.springframework.data.util.Pair;
+import org.springframework.stereotype.Service;
+
+import java.util.*;
+
+@Service
+public class PaymentOptionMapper {
+
+  private final InstallmentMapper installmentMapper;
+  private final InstallmentPIIMapper installmentPIIMapper;
+
+  public PaymentOptionMapper(InstallmentMapper installmentMapper, InstallmentPIIMapper installmentPIIMapper) {
+    this.installmentMapper = installmentMapper;
+    this.installmentPIIMapper = installmentPIIMapper;
+  }
+
+  public Pair<PaymentOption, Map<InstallmentNoPII, Installment>> mapToModel(PaymentOptionDTO dto) {
+    Map<InstallmentNoPII, Installment> installmentMapping = new HashMap<>();
+
+    List<Installment> installments = dto.getInstallments().stream()
+      .map(installmentMapper::mapToModel)
+      .toList();
+
+    SortedSet<InstallmentNoPII> installmentNoPIIs = installments.stream()
+      .map(installment -> {
+        Pair<InstallmentNoPII, InstallmentPIIDTO> result = installmentPIIMapper.map(installment);
+        InstallmentNoPII installmentNoPII = result.getFirst();
+        installmentMapping.put(installmentNoPII, installment);
+        return installmentNoPII;
+      })
+      .collect(() -> new TreeSet<>(Comparator.comparing(InstallmentNoPII::getInstallmentId)),
```

Comment:
InstallmentNotPII implements Comparable, it should not be required to provide a comparator
```suggestion
      .collect(() -> new TreeSet<>(),
```

Reply:
modified